os/board/rtl8721csm/src/rtl8721csm_mi48.c

```diff
@@ -0,0 +1,189 @@
+/****************************************************************************
+ *
+ * Copyright 2023 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <tinyara/irq.h>
+#include <tinyara/sensors/mi48.h>
+#include <debug.h>
+#include <errno.h>
+#include "gpio_api.h"
+#include "gpio_irq_api.h"
+
+#ifdef CONFIG_MI48
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+#define DEVNAME_FORMAT	"/dev/img%d"
+
+#define GPIO_RESET		PA_0	// will set it to the right pin after wards
```

Comment:
so these pin numbers are only things related to rtl8721.
Then we can change init function as below, then we can merge this file with mi48.c
```
/** doxygen to explain each of params ...**/
int mi48_register(FAR const char *devname, FAR struct mi48_lower_s *lower, FAR struct i2c_dev_s *i2c, FAR struct spi_dev_s *spi, gpio_t pin1, gpio_t pin2) // but reset is necessary?
{
    ...
   gpio_irq_init(&(g_rtl8721csm_dev.gpio_info.data_ready), pin1)
   ...
}

Reply:
gpio initializations, gpio control and attaching irq handlers for gpio interrupts are all board specific logic, so I wanted to keep it seperate from the sensor code, which is common for all boards.